### PR TITLE
Migrate announcement component to React

### PR DESCRIPTION
**Pull request by anirudhnigania**
The pull request has been segregated into 5 Commits - 

1. 0347439 - Announcement View and Announcement Controller
2. bd9fc7b - API for Announcement
3. 79e1ef6 - Announcement Index Page
4. 5e507bd - Announcement Edit/ New Form, Announcement Action Buttons
5. d3e6515 - Announcement Actions and Reducers.

**Announcement Index Page**
announcements_index_page

**Announcement Edit Page **
announcement_edit_page

**Announcement New Page**
announcement_new_page

### CODE REVIEW

**Comment by jeremyyap** [comment]
Could you edit your commit messages? The first line should be a short summary of the changes. Detailed information can be included from the third line onwards. Also, you should use the imperative mood in the first line, e.g. 

`Announcement component migration to react` -> `Migrate announcement component to React`

https://chris.beams.io/posts/git-commit/

**Comment by anirudhnigania** [comment]
I have edited the commit messages and fixed the hound violations.

**Comment by allenwq** [comment]
@anirudhnigania Have you considered pagination in the index page? That is when there are too many announcements, we should split them page by page.  It's an optional task but the API and the actual page structure must take that into consideration. 

😄 Given that I would suggest you to PR the index page separately, and leave the new and edit page as it is first.

File: app/views/course/announcements/announcement.json.jbuilder [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+  json.linkToUser user_path(@announcement.creator)
+
+  json.formatedDateTime format_datetime(@announcement.start_at)
+  json.timePeriodMessage time_period_message(@announcement)  
+
+  json.canEdit can?(:edit, @announcement)
+  json.canDelete can?(:delete, @announcement)
+
+  json.currentlyActive @announcement.currently_active?
+end
```
Comment (fonglh) [inline app/views/course/announcements/announcement.json.jbuilder]: missing newline

File: app/views/course/announcements/index.json.jbuilder [hunk 2]
```diff
@@ -0,0 +1,26 @@
+json.announcements @announcements do |announcement| 
+  json.title announcement.title
```
Comment (fonglh) [inline app/views/course/announcements/index.json.jbuilder]: Looks nearly identical to `announcement.json.jbuilder`. Can it be refactored into a partial?

**Pull request closed without merging**